app/react/portainer/settings/SettingsView/SSLSettingsPanel/SSLSettingsPanel.tsx

```diff
@@ -135,16 +133,16 @@ function SSLSettingsPanel() {
   );
 
   function handleSubmit({ certFile, forceHTTPS, keyFile }: FormValues) {
-    if (!certFile || !keyFile) {
-      return;
-    }
+    // certFile and keyFile can be null, because if the forceHTTPS is
```

Comment:
this is incorrect. server generates certificates only on init. 
in ssl_update, if certFile is null, it will ignore and continue to setting other settings (such as forceHTTPs). see api/http/handler/ssl/ssl_update.go

Reply:
When `forceHTTPs` toggle is on, it doesn't mean that server needs to generate the new self-signed certificates, instead, it disables the http connection. That is, when `forceHTTPs` is off, the server can be accessed by both http and https. when `forceHTTPs` is on, only https can access the server. 
> if certFile is null, it will ignore and continue to setting other settings (such as forceHTTPs). 
I think this is expected behavior.

Reply:
exactly, that's why the comment is incorrect. the self generated certs are generated only at the start of the server. a user always have certs, either his own supplied, or self generated.